src/netlink.h

```diff
@@ -24,11 +26,17 @@ typedef int t_pid;
 typedef int t_groups;
 typedef int t_type;
 
+struct nlmsgbuf{
+    struct nlmsghdr hdr;
+    char msg[NLMSG_ALIGN(MAX_PAYLOAD)] __attribute__((aligned(NLMSG_ALIGNTO)));
+};
+
 typedef struct {
     t_socket fd; 
     t_timeout tm;  
     t_pid srcpid; 
     t_type type; 
+    struct nlmsgbuf *nlb;
```

Comment:
Why this needs to be a pointer?

Reply:
No reason I suppose, we could even inline `nlmsgbuf` entirely at this point. It felt natural to me to keep it separate but I suppose the indirection is a waste since this all gets heap-allocated anyway.

(Sorry for the late reply, first I was sick and then I got swamped with some other stuff.)